// @udecode/plate-mention/dist/react/index.mjs
// src/react/MentionPlugin.tsx
import { toPlatePlugin } from "@udecode/plate/react";

// src/lib/BaseMentionPlugin.ts
import {
  createSlatePlugin,
  createTSlatePlugin
} from "@udecode/plate";
import {
  withTriggerCombobox
} from "@udecode/plate-combobox";
var BaseMentionInputPlugin = createSlatePlugin({
  key: "mention_input",
  node: { isElement: true, isInline: true, isVoid: true }
});
var BaseMentionPlugin = createTSlatePlugin({
  key: "mention",
  node: { isElement: true, isInline: true, isMarkableVoid: true, isVoid: true },
  options: {
    trigger: "@",
    triggerPreviousCharPattern: /^\s?$/,
    createComboboxInput: (trigger) => ({
      children: [{ text: "" }],
      trigger,
      type: BaseMentionInputPlugin.key
    })
  },
  plugins: [BaseMentionInputPlugin]
}).extendEditorTransforms(({ editor, type }) => ({
  insert: {
    mention: ({ key, value }) => {
      editor.tf.insertNodes({
        key,
        children: [{ text: "" }],
        type,
        value
      });
    }
  }
})).overrideEditor(withTriggerCombobox);

// src/react/MentionPlugin.tsx
var MentionPlugin = toPlatePlugin(BaseMentionPlugin);
var MentionInputPlugin = toPlatePlugin(BaseMentionInputPlugin);
export {
  MentionInputPlugin,
  MentionPlugin
};
